### PR TITLE
langgraph checkpoint: new library for checkpoint interfaces

### PR DESCRIPTION
**Pull request by vbarda**

### CODE REVIEW

File: libs/checkpoint/tests/checkpoint/test_memory.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
+from datetime import datetime, timezone
```
Comment (nfcampos) [inline libs/checkpoint/tests/checkpoint/test_memory.py:1]: remove this checkpoint folder, no need to make things harder to find

File: docs/docs/reference/checkpoints.md [hunk 2]
```diff
@@ -1,35 +1,19 @@
...
 
-LangGraph also natively provides the following checkpoint implementations.
-
-### MemorySaver
+LangGraph natively provides the following checkpoint implementations:
 
-::: langgraph.checkpoint.memory.MemorySaver
+### MemorySaver (via `langgraph_checkpoint`)
 
 ### AsyncSqliteSaver
```
[the remark this answers is not in the document]
Comment (vbarda) [reply docs/docs/reference/checkpoints.md:18]: yea that was my plan, now that we've moved i'll update as well!

File: libs/checkpoint/README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+Each checkpointer should conform to `langgraph_checkpoint.BaseCheckpointSaver` interface and must implement the following methods:
+
+- `.put` - Store a checkpoint with its configuration and metadata.
+- `.put_writes` - Store intermediate writes linked to a checkpoint (i.e. pending writes).
+- `.get_tuple` - Fetch a checkpoint tuple using for a given configuration (`thread_id` and `thread_ts`).
+- `.list` - List checkpoints that match a given configuration and filter criteria.
+
+If the checkpointer will be used with asynchronous graph execution (i.e. executing the graph via `.ainvoke`, `.astream`, `.abatch`), checkpointer must implement asynchronous versions of the above methods (`.aput`, `.aput_writes`, `.aget_tuple`, `.alist`).
+
+## Usage
```
[the remark this answers is not in the document]
Comment (vbarda) [reply libs/checkpoint/README.md:44]: sounds good

File: pyproject.toml [hunk 4]
```diff
@@ -11,6 +11,7 @@ python = "^3.10"
 
 [tool.poetry.group.docs.dependencies]
 langgraph = { path = "libs/langgraph/", develop = true }
+langgraph-checkpoint = { path = "libs/checkpoint/", develop = true }
```
Comment (nfcampos) [inline pyproject.toml:14]: what is this one used for?

Comment (vbarda) [reply pyproject.toml:14]: just for docs building (so that the references for checkpointers are valid)

**Pull request merged**